wafer/kv/serializers.py

```diff
@@ -1,8 +1,57 @@
 from django.core.exceptions import PermissionDenied
+from django.core.urlresolvers import NoReverseMatch
 from rest_framework import serializers
 from wafer.kv.models import KeyValue
 
 
+class MaybeHyperlinkField(serializers.HyperlinkedRelatedField):
+    """Overrider HyperlinkRelatedFiled to return just the object name
+       for when a model has not been hooked up to the api, so we don't
+       need to add special cases everytime a new model grows KV support
+       without growing a API view."""
+
+    def get_url(self, obj, view_name, request, format):
+        try:
+            return super(MaybeHyperlinkField, self).get_url(obj, view_name, request, format)
+        except NoReverseMatch:
+            return super(MaybeHyperlinkField, self).get_name(obj)
+
+
+class CustomKeyValueSerializer(serializers.HyperlinkedRelatedField):
+
+    # Because of how Django rest framework handles the dance between
+    # ManyRelatedField and RelatedField, we must override __new__
+    # to have this always behave as if 'many=True' was set
+    def __new__(cls, *args, **kwargs):
+        if 'view_name' not in kwargs:
+            # queryset here is just to statisfy older versions of
+            # rest-framework which don't check for overridding get_queryset.
+            return cls.many_init(*args, view_name='keyvalue-detail',
+                                 queryset=KeyValue.objects.none(), **kwargs)
+        return super(CustomKeyValueSerializer, cls).__new__(cls, *args,
+                                                            **kwargs)
+
+    def get_queryset(self):
+        """Custom queryset to limit key-value pairs returned to the
+           requesting user's group for choices"""
+        request = self.context.get('request', None)
+        if request.user.id is not None:
```

Comment:
Should this be `if request and request.user.id is not None:` to handle the case where there is no request?